augur/io/metadata.py

```diff
@@ -475,3 +475,6 @@ def _get_delimiter(path: str, valid_delimiters: Iterable[str]):
             # This assumes all csv.Errors imply a delimiter issue. That might
             # change in a future Python version.
             raise InvalidDelimiter from error
+        except UnicodeDecodeError:
+            # FIXME: add offending byte character in error output and suggest remediation steps
+            raise AugurError(f"File is not encoded as '{ENCODING}'.")
```

Comment:
I'd suggest using the `encoding` attribute of the raised `UnicodeDecodeError` object instead of `ENCODING` here, to avoid unexpected mismatches (which would be very confusing to debug).

Reply:
Updated in [force-push](https://github.com/nextstrain/augur/compare/e2440099ef128c963cd4fea840a523867fd490c2..f263054a379e20e80161bd921a4d910733c5bcb5):

```py
except UnicodeDecodeError as e:
    raise AugurError(dedent(f"""\
        File '{path_or_buffer}' contains {e.object[e.start:e.end]} which is not encoded as '{e.encoding}'.
        Try saving the file using '{e.encoding}' encoding."""))
```

Reply:
Rather than manually surround interpolated values with single quotes, I'd recommend using the `!r` [conversion flag](https://docs.python.org/3/library/string.html#format-string-syntax) so that a unambiguous-in-all-circumstances representation is used. I'd do this for the `e.object[…]` interpolation too, even though that's omitting single quotes currently (presumably because a `bytes` object that's stringified produces `b'…'` already).

Reply:
[Done](https://github.com/nextstrain/augur/compare/9325ca99ef6d6295ffd85aa35044ea8330784033..cd479fdabceb31df9f0dff8fa63144b640cde96b)